### PR TITLE
Intellij spring boot support

### PR DESCRIPTION
**Pull request by jdneo**
Intellij spring boot support

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java [hunk 1]
```diff
@@ -61,105 +63,81 @@
...
-            processHandler.setText(UPLOADING_SUCCESSFUL);
+            FTPClient ftp = WebAppUtils.getFtpConnection(profile);
+
+            uploadWebConfigFile(ftp);
+
+            int indexOfDot = webAppSettingModel.getTargetName().lastIndexOf(".");
+            String fileName = webAppSettingModel.getTargetName().substring(0, indexOfDot);
+            String fileType = webAppSettingModel.getTargetName().substring(indexOfDot);
+
+            uploadArtifact(input, webApp, ftp, fileName, fileType);
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java]: close input when there is exception

Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java]: Better define input closer to the usage place

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java [hunk 2]
```diff
@@ -61,105 +63,81 @@
...
-    private static final String NO_TARGETFILE = "Cannot find target file: %s";
-    private static final String FAIL_FTPSTORE = "FTP client can't store the artifact, reply code: %s";
-    private static final String BASE_PATH = "/site/wwwroot/webapps/";
+    private static final String DEPLOY_SUCCESSFUL = "Deploy successfully!";
+    private static final String STOP_DEPLOY = "Deploy Failed!";
+    private static final String NO_WEB_APP = "Cannot get webapp for deploy.";
+    private static final String NO_TARGET_FILE = "Cannot find target file: %s.";
+    private static final String FAIL_FTP_STORE = "FTP client can't store the artifact, reply code: %s.";
+
+    private static final String WEB_CONFIG_PACKAGE_PATH = "/com/microsoft/azuretools/core/mvp/model/webapp/web.config";
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java]: put template file to resource

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java [hunk 3]
```diff
@@ -178,6 +156,92 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
...
+        if (webAppSettingModel.isCreatingNew() && Comparing.equal(webAppSettingModel.getDeploymentType(),
+                WebAppSettingModel.DeploymentType.SPRING_BOOT)) {
+            processHandler.setText(UPLOADING_WEB_CONFIG);
+            InputStream webConfigInput = getClass()
+                    .getResourceAsStream(WEB_CONFIG_PACKAGE_PATH);
+            uploadFileToFtp(ftp, WEB_CONFIG_FTP_PATH, webConfigInput);
+        }
+    }
+
+    private void uploadArtifact(FileInputStream input, WebApp webApp, FTPClient ftp, String fileName, String fileType)
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java]: add notnull

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java [hunk 4]
```diff
@@ -178,6 +156,92 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
...
+                    .getWebAppById(webAppSettingModel.getSubscriptionId(), webAppSettingModel.getWebAppId());
+        }
+        if (webApp == null) {
+            processHandler.setText(STOP_DEPLOY);
+            throw new Exception(NO_WEB_APP);
+        }
+        return webApp;
+    }
+
+    private void uploadWebConfigFile(FTPClient ftp) throws IOException {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java]: notnull?

**Comment by Eskibear** [comment]
retest this please

**Pull request merged**